### PR TITLE
Reformat discovery response to match expected format.

### PR DESCRIPTION
**Pull request by aroberson**

### CODE REVIEW

**Comment by aroberson** [comment]
Please merge with https://github.com/dell-asm/asm-core/pull/3461

**Comment by AlexDron** [comment]
@gavin-scott or @sushilrai please merge, I don't have privs.

File: bin/discovery.rb [hunk 1]
```diff
@@ -28,23 +28,34 @@ def collect_scaleio_facts
...
-    facts[:protection_domains] << pd
+    protection_domain[:sds_list] << sds
   end
+  scaleio_volumes(scaleio_system).each do |volume|
+    storage_pool = facts[:protection_domain_list].find do |protection_domain|
+      protection_domain[:storage_pool_list].find do |storage_pool|
+        storage_pool[:id] == volume[:storagePoolId]
+      end
+    end
+    storage_pool[:volume_list] << volume
```
Comment (sushilrai) [inline bin/discovery.rb]: Discovery operation failed at this step for the ScaleIO Gateway where volumes are created.

Comment (sushilrai) [reply bin/discovery.rb]: Updated discovery script is available at https://gist.github.com/sushilrai/81a12c5854b8da10d0eb0af57eceae04

**Comment by aroberson** [comment]
@gavin-scott ready for merge

**Pull request merged**